buildspec/linuxIntegrationTests.yml

```diff
@@ -28,6 +28,17 @@ phases:
 
     pre_build:
         commands:
+            # If present, log into CodeArtifact. Provides a nice safety net in case NPM is down.
+            # Should only affect tests run through IDEs team-hosted CodeBuild.
+            - |
+                if [ $TOOLKITS_CODEARTIFACT_DOMAIN ] && [ $TOOLKITS_CODEARTIFACT_REPO ] && [ $ACCOUNT_ID ]; then
+                  aws codeartifact login --tool npm --domain "$TOOLKITS_CODEARTIFACT_DOMAIN" --domain-owner "$ACCOUNT_ID" --repository "$TOOLKITS_CODEARTIFACT_REPO" > /dev/null 2>&1
+                  if [ $? -eq 0 ]; then
+                    echo "Connected to CodeArtifact"
+                  else
+                    echo "CodeArtifact connection failed. Falling back to npm"
+                  fi
+                fi
```

Comment:
Please run this block through https://www.shellcheck.net/ . It has several issues. E.g. missing quotes around env vars.

Reply:
Got a Powershell equivalent? 😛

Reply:
:) Not sure, but powershell has many fewer gotchas.